### ISSUE TITLE
Rename onpremise to self-hosted

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/self-hosted#796]
In the [docs](https://develop.sentry.dev/self-hosted/) we're using "self-hosted" to name what we're doing here. For branding purposes I would like to standardize on that and stop referring to "onpremise."

(Short URL, fwiw: https://git.io/JSDlX)

### Todo

- [x] rename repo
- [x] [code](https://github.com/search?q=org%3Agetsentry+onpremise&type=code)
  - [x] easy
      - [x] `sentry.io`—archived repo
      - [x] `sentry-release`—moribund repo, don't care
      - [x] https://github.com/getsentry/labels/commit/9d35accca4af33a52ab2cbb91b65a33adc077deb
      - [x] https://github.com/getsentry/deps/commit/5962f6b603d73095b9e72ca68e417f15a7094d78
      - [x] https://github.com/getsentry/demo/pull/11
      - [x] https://github.com/getsentry/develop/pull/469
      - [x] https://github.com/getsentry/eng-pipes/pull/273
      - [x] https://github.com/getsentry/etl/pull/671
      - [x] https://github.com/getsentry/lookml/pull/318
      - [x] https://github.com/getsentry/publish/pull/670
      - [x] https://github.com/getsentry/relay/pull/1136
      - [x] https://github.com/getsentry/self-hosted/pull/1169
      - [x] `sentry`
          - [x] https://github.com/getsentry/sentry/pull/30482
          - [x] https://github.com/getsentry/sentry/pull/30483
          - [x] https://github.com/getsentry/sentry/pull/30486
      - [x] https://github.com/getsentry/sentry-dart/pull/642
      - [x] https://github.com/getsentry/sentry-docs/pull/4437
      - [x] https://github.com/getsentry/snuba/pull/2240
      - [x] https://github.com/getsentry/static-sites/pull/773
  - [x] [hard](https://github.com/getsentry/self-hosted/issues/796#issuecomment-990355748)
    - [x] backend
      - [x] https://github.com/getsentry/getsentry/pull/6841
      - [x] https://github.com/getsentry/single-tenant/pull/429
      - [x] https://github.com/getsentry/sentry/pull/30973
      - [x] https://github.com/getsentry/sentry/pull/31066
    - [x] frontend
      - [x] add `isSelfHosted` in `sentry` (done in [#30973](https://github.com/getsentry/sentry/pull/30973))
      - [x] https://github.com/getsentry/sentry/pull/31065
    - [x] trust it
      - [x] https://github.com/getsentry/getsentry/pull/6866
      - [x] ~~remove `SENTRY_ONPREMISE` in `single-tenant`~~
        - [x] punting for "months" ... documented in https://github.com/getsentry/single-tenant/pull/432
    - [x] remove `SENTRY_ONPREMISE` and `isOnPremise` in `sentry` (kill backcompat) – do this, like, waaaaay later
- [x] [forums](https://forum.sentry.io/c/on-premise/7)—https://github.com/getsentry/onpremise/issues/1151
- [x] [~blog~](https://www.google.com/search?q=%22on-premise%22+site%3Ahttps%3A%2F%2Fblog.sentry.io)—https://github.com/getsentry/self-hosted/issues/796#issuecomment-978058356
- [x] https://github.com/getsentry/eng-pipes/issues/274
- [x] Audit to make sure monthly CalVer release will work.

**Comment by BYK** [comment]
I'm not sure if the effort is worth the benefits.

**Comment by chadwhitacre** [comment]
GitHub handles renames gracefully. The one lingering question I've had is forks, but it seems from testing (not seeing docs) that there's no issue with a fork of a renamed repo. The fork keeps the old name, _and_ its connection to the renamed parent. If anyone with a fork has the upstream set as a remote, GitHub redirects/aliasing ensure that they are unaffected.

tl;dr I think the effort is trivial.

### Fork Test

<img width="843" alt="Screen Shot 2021-01-06 at 4 54 56 PM" src="https://user-images.githubusercontent.com/134455/103823479-2a230800-5040-11eb-8411-3755868910a8.png">

**Comment by BYK** [comment]
Alright, you know what, let's try this :)

**Comment by chadwhitacre** [comment]
@ingria Do you foresee problems with this? Why are you 😕?

**Comment by ingria** [comment]
Technically - no, not at all. I just didn't understand what the benefits of renaming would be. That's why I was a little :confused:

Anyway, there are also some other places that might need to be edited:

- Help articles ([example](https://help.sentry.io/hc/en-us/articles/115006979128))
- [Community forums](https://forum.sentry.io/c/on-premise/7)
- Blog articles ([example](https://blog.sentry.io/2016/01/08/sentry-8-on-premise))

**Comment by chadwhitacre** [comment]
> benefits

Less confusion. Right now we use two names—"on-premise," "self-hosted"—for the same thing. May not be worth it. Need to stew on this some more ... 🤔

**Comment by chadwhitacre** [comment]
The more I think about it, the more I don't like either name—`onpremise` or `self-hosted`—for this repo. I think a better name would be `single-node`, because it better captures our intention here: to package up _all_ of Sentry for deployment on a single node, in order to support a) proofs of concept for people evaluating SaaS, and b) small deployments for students, hobbyists, and small orgs. "Self-hosted" is broader as it really encompasses multi-node setups, too, which we don't officially support and never will ([SaaS](https://sentry.io/pricing/)! [SaaS](https://sentry.io/pricing/)!). Moreover, work that is relevant to self-hosters of any scale quite often (almost always?) impacts our other "primary" repos, e.g. #863 and #857 should really be against `sentry`, imo. We have a label in that repo to flag issues of particular (exclusive?) relevance to self-hosters.

**Comment by ingria** [comment]
This name may better reflect _how_ it's supposed to work, but it doesn't describe _what_ is this project for.

I mean, imagine you've just heard about some cool project called **Gentry**. And you'd like to know whether it's possible to use it on your own servers. How would you google it: "gentry self hosted" or "gentry single node"?

In other words, "single-node" is ok for a repo name, but not ok for a product name. Just my opinion, of course.

**Comment by BYK** [comment]
I agree with @ingria here, this was my main concern too. How about something like `self-hosted-core` or something as this setup _can_ be expanded into a multi-node setup through Docker Swarm or something like https://kompose.io/. Also AFAIK https://github.com/sentry-kubernetes/charts/ is tracking this repo for updates.

**Comment by chadwhitacre** [comment]
Returning to the original idea here, renaming to `self-hosted`.

**Comment by chadwhitacre** [comment]
I ran this by the Backend TSC and nobody raised any red flags from a technical standpoint, experience with repo renames is that GitHub does a great job making it seamless. All clear from the standpoint!

**Comment by chadwhitacre** [comment]
> Help articles ([example](https://help.sentry.io/hc/en-us/articles/115006979128))

> [Community forums](https://forum.sentry.io/c/on-premise/7)

I just started a thread in the forums to [see if anyone is paying attention](https://forum.sentry.io/t/anyone-out-there-self-hosted-community-meta/15559) there. Where I want to take that conversation is I actually want to deprecate the forum in favor of GitHub Issues. 🐭

> Blog articles ([example](https://blog.sentry.io/2016/01/08/sentry-8-on-premise))

I do find [references to on-premise](https://www.google.com/search?q=%22on-premise%22+site%3Ahttps%3A%2F%2Fblog.sentry.io) historically on the blog, looks like [June, 2020](https://blog.sentry.io/2020/06/22/self-hosted-sentry-switching-to-calver) is the latest? Not all of the references

**Comment by chadwhitacre** [comment]
I added a todo to this ticket. I'm going to let [this forum meta post](https://forum.sentry.io/t/anyone-out-there-self-hosted-community-meta/15559) sit for a day or two, if it's 🦗 s then I intend to proceed with this ticket. If it's not then we'll talk first. :)

**Comment by chadwhitacre** [comment]
I'm planning to rename the repo next week during the US Thanksgiving lull.

**Comment by chadwhitacre** [comment]
Repo is done: https://github.com/getsentry/self-hosted. 💃

**Comment by chadwhitacre** [comment]
After reviewing the blog posts for mentions of "on-premise," I don't think we need to go back through and change those. There are 11 mentions but one is in reference to a different product. Of the 10 posts that talk about on-premise Sentry, three also refer to it as "open source" Sentry, which is definitely something we are moving away from—SaaS is open-source too, there's no feature difference between SaaS, single-tenant, and self-hosted. All that said, historically we did use "on-premise" and "open-source" to refer to the product now called "self-hosted Sentry," and I don't think we need to rewrite history. Being consistent in the future is enough.

1. https://blog.sentry.io/2020/06/22/self-hosted-sentry-switching-to-calver (1)
1. https://blog.sentry.io/2020/01/07/self-hosted-sentry-10-is-ready-to-serve-get-it-while-its-hot (1)
1. https://blog.sentry.io/2019/03/14/nextcloud-uses-sentry-to-build-federated-clouds (1) — this one refers to Nextcloud, not Sentry
1. https://blog.sentry.io/2019/02/14/sentry-thrives-open-source-software-company (3 + "open source")
1. https://blog.sentry.io/2016/07/05/releasing-sentry (4 + "open source")
1. https://blog.sentry.io/2016/01/12/monitoring-the-monitor (8)
1. https://blog.sentry.io/2015/12/29/sentry-js-sdk-release (1)
1. https://blog.sentry.io/2015/10/08/sentry-8-is-here (8)
1. https://blog.sentry.io/2015/09/29/sso-for-all (8)
1. https://blog.sentry.io/2015/08/04/rethinking-sentrys-documentation (3 + "open source")
1. https://blog.sentry.io/2015/06/30/driven-by-open-source (1)

**Comment by chadwhitacre** [comment]
I audited [our code](https://github.com/search?q=org%3Agetsentry+onpremise&type=code), kicked out a bunch of PRs, and noted the remaining work in the TOOD in the description.

**Comment by chadwhitacre** [comment]
I made PRs for the rest of the repos except for `sentry` and `getsentry`. Those will take some more thought as what is at issue is a configuration change and we need to do handle that in a graceful way.

**Comment by renchap** [comment]
As the forum topic is closed, I will reply here.
I suggest moving to Github Discussions, not Issues, as they provide a much better platform for this kind of… discussions, with useful tools like marking topics as solved, accepted answers, threading and such.

It is easy to convert an issue to a discussion (or the other way around), and it does not clutter the issue tracker with non-issue things, and (if i remember correctly) you can redirect people to open a discussion instead of an issue in the configurable "new issue" page.

As for the Github are really improving issues and discussions a lot and the OSS community is switching to it quickly, so I assume many people will be familiar with GH Discussions in the coming months and it wont really be an issue.

**Comment by chadwhitacre** [comment]
@renchap Moving the discussion re: Discussions to https://github.com/getsentry/self-hosted/issues/1151#issuecomment-982066990. ;-)

**Comment by chadwhitacre** [comment]
> Audit to make sure monthly CalVer release will work.

I got nervous when I noticed that the publish repo [looks at the issue title](https://github.com/getsentry/publish/pull/670/files), but we should be fine because the issue title [comes from the repo slug](https://github.com/getsentry/action-prepare-release/blob/fa92eb5b15e20315c39d1da48a502d4f7bc73190/action.yml#L92). 👍

**Comment by chadwhitacre** [comment]
@ayozemr @fliespl et al. Now that #1169 has landed would someone be able to upgrade to the latest `master` and report back if you encounter any issues? I'd love to beta-test this a bit before the next monthly release. 🙏

**Comment by chadwhitacre** [comment]
(Inlining ...)

The better way to do this is to:

1. Set `SENTRY_SELF_HOSTED` in the downstream apps (`getsentry` and `single-tenant`) as a no-op duplicate. It sits right next to `SENTRY_ONPREMISE` but nothing reads it.
2. Modify the upstream (i.e., `sentry`, here) to use/read `SENTRY_SELF_HOSTED` on the backend (basically this here PR here).
3. Remove `SENTRY_ONPREMISE` from the downstreams.
4. Modify the upstream to set `isSelfHosted` as a no-op duplicate on the frontend.
5. Modify the downstreams to read from `isSelfHosted`.
6. Modify the upstream to read from `isSelfHosted` and remove `isOnPremise`.

The crux is that downstreams _set_ the backend `SENTRY_FOO` but _consume_ the frontend `isFooBar`, while upstreams consume the backend and set the frontend. The right approach looks like this:

```
backend  add    downstream
backend  add    upstream
backend  remove upstream
backend  remove downstream

frontend add    upstream
frontend add    downstream
frontend remove downstream
frontend remove upstream
```

**Comment by chadwhitacre** [comment]
- don't remove ONPREMISE from sentry (upstream) until after removed from downstream
- check on single-tenant flow, account for need to cut a release

**Comment by chadwhitacre** [comment]
Big config changes shipped to SaaS prod. A couple more bits to flip in the coming weeks/months and then removing backcompat "at some point."

**Comment by chadwhitacre** [comment]
Flipped the bit on `getsentry`, punting indefinitely on `single-tenant`. Gonna declare victory here.

Rename: check! 💃
